sdk-extensions/autoconfigure-spi/src/main/java/io/opentelemetry/sdk/autoconfigure/spi/AutoConfigurationCustomizer.java

```diff
@@ -56,6 +56,9 @@ AutoConfigurationCustomizer addSamplerCustomizer(
    * allow customization. The return value of the {@link BiFunction} will replace the passed-in
    * argument.
    *
+   * <p>NOTE: If returning a different exporter instance, be sure to call {@link
+   * SpanExporter#shutdown()} on the instance passed as an argument to release resources.
```

Comment:
I think it's common to use this method to decorate the original

what do you think of putting this warning on `toBuilder()` instead

Reply:
Is delegation a common pattern for this as well, in which case calling `shutdown()` would break the inner/underlying implementation right?

Reply:
The `toBuilder()` methods already have this warning, e.g. [OtlpGrpcSpanExporter#toBuilder](https://github.com/open-telemetry/opentelemetry-java/blob/1b9855748b05754c4707ab876b8484b816997af3/exporters/otlp/all/src/main/java/io/opentelemetry/exporter/otlp/trace/OtlpGrpcSpanExporter.java#L55). 

Was trying the belt and suspenders approach, but maybe unnecessary. How about I close this and re-open it if we get issues with resources not being cleaned up?